x/evm/keeper/opcodes_hooks.go

```diff
@@ -0,0 +1,37 @@
+// Copyright Tharsis Labs Ltd.(Evmos)
+// SPDX-License-Identifier:ENCL-1.0(https://github.com/evmos/evmos/blob/main/LICENSE)
+
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+type DefaultOpCodesHooks struct {
+	accessControl PermissionPolicy
+	signer        string
+}
+
+func NewDefaultOpCodesHooks(accessControl PermissionPolicy, signer string) vm.OpCodeHooks {
+	return &DefaultOpCodesHooks{
+		accessControl: accessControl,
+		signer:        signer,
+	}
+}
+
+func (h *DefaultOpCodesHooks) CreateHook(_ *vm.EVM, caller common.Address) error {
```

Comment:
is this hook executed before or after the CREATE? It's not clear and it also doesn't have a comment